org.eclipse.xtext/src/org/eclipse/xtext/xtext/XtextValidator.java

```diff
@@ -1255,6 +1255,11 @@ public void checkOverridingRule(AbstractRule rule) {
 		final boolean isOverride =
 				rule.getAnnotations().stream().anyMatch(e -> AnnotationNames.OVERRIDE.equals(e.getName()));
 		
+		if (GrammarUtil.getGrammar(rule).getUsedGrammars().isEmpty() && isOverride) {
```

Comment:
Nitpicking: Why do you dublicate `GrammarUtil.getGrammar(rule).getUsedGrammars()` instead of using a local var? It would also make sense to guard with `if (isOverride)` before doing the other stuff. Both are minor things, though.

Reply:
I will add the local variable, but I am not sure what you mean with "the other stuff" that should be guarded by `isOverride`. The for-loop that comes after my contribution contains a case for `!isOverride`, so such a guard does not apply to it.